R/epGRanges.R

```diff
@@ -0,0 +1,46 @@
+## Equipartition GRanges:
+## Splits a GRanges object into a specified number of parts with equal (total) width.
+## Each part is a GRanges and all parts are returned as GRangesList object. 
+
+## Alexandre Kuhn (alexandre.m.kuhn@gmail.com)
+
+epGRanges<-function(gr,partitionNb) {
+        if (partitionNb<1) stop("The number of partition should be at least 1.")
+        else if (partitionNb==1) return(GRangesList(gr))
+        else
+        {
+        grLength<-length(gr)
+        cs<-cumsum(width(gr))
+
+        partitionWidth<-floor(cs[grLength]/partitionNb) #last partition is longer
+        #determine partitions' upper bounds (ub)
+        upperbound<-seq(partitionWidth,cs[grLength],partitionWidth)
```

Comment:
`IRanges::breakInChunks()` is probably a more convenient way to do this. It returns a formal _Partitioning_ object, which may be helpful.